app/migrations/Version20201207114926.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * @copyright   <year> Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        https://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\Exception\SkipMigration;
+use Mautic\CoreBundle\Doctrine\AbstractMauticMigration;
+
+final class Version20201207114926 extends AbstractMauticMigration
+{
+    /**
+     * @throws SkipMigrationException
+     */
+    public function preUp(Schema $schema): void
+    {
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql("UPDATE {$this->prefix}lead_fields SET is_unique_identifer = 0 WHERE object = 'company';");
+
+        $this->addSql("UPDATE {$this->prefix}lead_fields SET is_unique_identifer = 1 WHERE object = 'company' and alias in ('companyname');");
```

Comment:
By default is enabled companyname and companyemail in custom contact fields (but not use it).
In company import use company with combination of city/state/country field.
Other parts in Mautic use IdentifyCompanyHelper just with companyname to identify already existed company.


Then I decide use companyname as unique by default. It fit current behaviour, it was never documented, and nobody knows how it works.

Reply:
Makes sense! Would you mind adding it to `UPGRADE-4.0.md` so that folks are aware about it when upgrading? 😊 https://github.com/mautic/mautic/blob/features/UPGRADE-4.0.md